Lib/dbm/sqlite3.py

```diff
@@ -60,18 +60,22 @@ def __init__(self, path, /, *, flag, mode):
         # We use the URI format when opening the database.
         uri = _normalize_uri(path)
         uri = f"{uri}?mode={flag}"
+        if flag == "ro":
+            # Add immutable=1 to allow read-only SQLite access even if wal/shm missing
+            uri += "&immutable=1"
```

Comment:
```suggestion
            uri = f"{uri}&immutable=1"
```

Reply:
Why?

Reply:
Like in line 62; `uri = f"{uri}?mode={flag}"`

Reply:
I think it's insignificant.